modules/core/02-client/keeper/events.go

```diff
@@ -24,13 +26,19 @@ func EmitCreateClientEvent(ctx sdk.Context, clientID string, clientState exporte
 }
 
 // EmitUpdateClientEvent emits an update client event
-func EmitUpdateClientEvent(ctx sdk.Context, clientID string, clientType string, consensusHeight exported.Height, clientMsgStr string) {
+func EmitUpdateClientEvent(ctx sdk.Context, clientID string, clientType string, consensusHeights []exported.Height, clientMsgStr string) {
+	var consensusHeightStr []string
+	for _, height := range consensusHeights {
+		consensusHeightStr = append(consensusHeightStr, height.String())
+	}
+
 	ctx.EventManager().EmitEvents(sdk.Events{
 		sdk.NewEvent(
 			types.EventTypeUpdateClient,
 			sdk.NewAttribute(types.AttributeKeyClientID, clientID),
 			sdk.NewAttribute(types.AttributeKeyClientType, clientType),
-			sdk.NewAttribute(types.AttributeKeyConsensusHeight, consensusHeight.String()),
+			sdk.NewAttribute(types.AttributeKeyConsensusHeight, consensusHeights[0].String()),
```

Comment:
This was being checked in the `UpdateClient()` method of keeper.go.

```
if len(consensusHeights) != 0 {
    // emit events, otherwise no-op & return nil
}

return nil
```

Reply:
ah yes. The only strange part is that this function assumes you pass in a non empty array of consensus heights

Unrelated, could we add `// deprecated` next to the `consensus height` attribute

Reply:
Should we backport `consensusHeights` attribute to older releases? That way relayers could switch to it without waiting for all chains to upgrade to this version

Reply:
> The only strange part is that this function assumes you pass in a non empty array of consensus heights

Agree, it is kind of strange. I'll adapt to check `len(consensusHeights)` inside event emission func and remove the check from the `UpdateClient` keeper method. 

> Should we backport consensusHeights attribute to older releases?

Sounds like a good idea, but this would require a separate PR to main, correct?

Reply:
good catch!